### PR TITLE
10-CORE-GET-api-users

### PR DESCRIPTION
**Pull request by miors**
endpoint /api/users
solution and tests added
API endpoint doc updated

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -285,3 +285,26 @@ describe("DELETE /api/comments/:comment_id", () => {
...
   });
 });
+
+describe("GET /api/users", () => {
+  it("GET:200 should return all users", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body: { users } }) => {
+        expect(users).toHaveLength(4);
```
Comment (pippaamy) [inline __tests__/integration.test.js:295]: lovely!!

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -285,3 +285,26 @@ describe("DELETE /api/comments/:comment_id", () => {
...
+          expect(typeof user.name).toBe("string");
+          expect(typeof user.avatar_url).toBe("string");
+        });
+      });
+  });
+
+  it("GET:404 should display error when route does not exist", () => {
+    return request(app)
+      .get("/api/notARoute")
+      .expect(404)
```
Comment (pippaamy) [inline __tests__/integration.test.js:307]: you may have a similar test to this from earlier with topics??

File: controllers/users.controller.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const { fetchAllUsers } = require("../models/users.model");
+
+exports.getAllUsers = (req, res, next) => {
+  fetchAllUsers()
+    .then((users) => {
+      res.status(200).send({ users });
```
Comment (pippaamy) [inline controllers/users.controller.js:6]: lovely!

File: models/users.model.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const db = require("../db/connection");
+const queries = require("../db/queries");
+
+exports.fetchAllUsers = () => {
+  return db
+    .query(`SELECT username, name, avatar_url FROM users;`)
+    .then(({ rows: users }) => {
```
Comment (pippaamy) [inline models/users.model.js:7]: love this deconstructing

**Review by pippaamy — COMMENTED** [review]
so fab, well done!

File: __tests__/integration.test.js (hunk 2 above)
Comment (miors) [reply __tests__/integration.test.js:307]: duplicate test removed

**Pull request merged**